benchmarks/tpcc/sysbench-tpcc.sh

```diff
@@ -529,22 +529,71 @@ function start_sysbench_containers()
+# This function watches the podman logs for all MySQL containers, to make sure they have all started before returning.
 # args: none
-# return: none
-function pause_for_stability()
-{
-    local seconds
+# return: 0=success, 1=error
+function pause_for_stability() {
+    for i in $(seq 1 ${PM_INSTANCES}); do
+        current_time=$(date '+%Y-%m-%d %H:%M:%S')
+        info_msg "Started waiting for container 'mysql${i}' to be healthy at ${current_time}"
+    done
 
-    seconds=300
-    total_seconds=$seconds
-    while [ $seconds -gt 0 ]; do
-        echo -ne "${STR_INFO} Pausing for $total_seconds seconds to give the containers time to initialize ... $seconds\033[0K\r"
-        sleep 1
-        seconds=$((seconds-1))
+    local unhealthy_containers=()
+    local elapsed_seconds=0
+    local all_healthy=false
+
+    while [ "$all_healthy" = false ]; do
+        all_healthy=true  # Assume all containers are healthy unless proven otherwise
+
+        for i in $(seq 1 ${PM_INSTANCES}); do
+            container_name="mysql${i}"
+            expected_message="X Plugin ready for connections. Bind-address: '::' port: 33060, socket: /var/run/mysqld/mysqlx.sock"
+
+            # Skip containers that have already been marked as unhealthy
+            if [[ " ${unhealthy_containers[@]} " =~ " ${container_name} " ]]; then
+                continue
+            fi
+
+            log_output=$(podman logs "$container_name" 2>&1 | tail -1)
+            container_status=$(podman inspect --format='{{.State.Status}}' "$container_name")
+
+            # Case(s) where the container is not in a good state
+            if [[ "$container_status" != "running" ]]; then
+                error_msg "Error: Container $container_name has exited or does not exist."
+                unhealthy_containers+=("$container_name")
+                all_healthy=false
+                continue
+            fi
+
+            # Container is healthy and running
+            if [[ "$log_output" == *"$expected_message"* ]]; then
+                info_msg "MySQL $container_name is ready."
+            else
+                all_healthy=false
+            fi
+        done
+
+        if [ "$all_healthy" = false ]; then
+            sleep 1
+            ((elapsed_seconds++))
+        fi
     done
-    echo
+
+    if [ ${#unhealthy_containers[@]} -ne 0 ]; then
+        error_msg "The following containers are not healthy:"
+        for container in "${unhealthy_containers[@]}"; do
+            error_msg "- $container"
+        done
+        error_msg "In order to do another fresh run, do the following steps (only 1 instance as an example, this benchmark run had $PM_INSTANCES):"
+        error_msg "- Stop and remove all sysbench containers, ex. 'podman stop sysbench1 && podman rm sysbench1'"
+        error_msg "- Remove all mysql containers, ex. 'podman rm mysql1'"
+        error_msg "- Delete the data volume mounted for all mysql containers, ex. 'sudo rm -rf /data/mysql1'"
+        return 1
```

Comment:
At this point should we get_container_logs and stop_containers?  the mysql* will be stopped and the sysbench* will be killed.

We will have capture any logs that will be needed to debug the problem.

Reply:
Done in [c7787d3](https://github.com/cxlbench/cxlbench/pull/28/commits/c7787d30e8d0527f25c55af5d1c8edbd90ba7e14)